lua/neo-tree/defaults.lua

```diff
@@ -38,6 +38,45 @@ local config = {
   sort_function = nil , -- uses a custom function for sorting files and directories in the tree
   use_popups_for_input = true, -- If false, inputs will use vim.ui.input() instead of custom floats.
   use_default_mappings = true,
+  source_selector = {
+    winbar = true, -- toggle to show selector on winbar
+    statusline = false, -- toggle to show selector on statusline
+    tab_labels = { -- falls back to source_name if nil
+      filesystem = " Files",
+      buffers = " Buffers",
+      git_status = " Git",
+    },
+    content_layout = "center", -- only with `tabs_layout` = "equal", "focus"
+    --                start  : |/ 裡 bufname     \/...
+    --                end    : |/     裡 bufname \/...
+    --                center : |/   裡 bufname   \/...
+    tabs_layout = "equal", -- start, end, center, equal, focus
+    --             start  : |/  a  \/  b  \/  c  \            |
+    --             end    : |            /  a  \/  b  \/  c  \|
+    --             center : |      /  a  \/  b  \/  c  \      |
+    --             equal  : |/    a    \/    b    \/    c    \|
+    --             active : |/  focused tab    \/  b  \/  c  \|
+    text_trunc_to_fit = true, -- toggle to truncate text in each tab when not enough width
```

Comment:
This is one of the easiest things to expose to the user, since I have to only change one line of the code. I thought more configuration is nicer.
A very niche case is that with `:set ambiwidth=double` (which I have never seen even a single person do this setting btw) the truncation sometimes go wrong with multibyte fonts.

![image](https://user-images.githubusercontent.com/41065736/179663005-6ebefcf2-374c-44ed-822f-ca7ed0c3d461.png)
![image](https://user-images.githubusercontent.com/41065736/179663104-ab6cab85-7330-4be9-8650-d26aa8c8b09c.png)

Reply:
I have hit this in my bufferline plugin re correctly truncating multibyte characters so you could borrow the function that does that if this would allow you to truncate without needing this option. Only skimmed the comment so this might not be that helpful

Reply:
Oh, somehow I didn't notice that you were the author of `bufferline.nvim`! Thanks a lot for the awesome plugin, I heavily depend on it everyday and keep up with your amazing work!
Yes, I do think there is a solution to it, but I didn't have the time and energy to go into this very niche edge-case. (and I got a bit lazy cuz neo-tree itself also has this problem.)

It would be so nice of you if you could fix this problem after this is merged into the `main` branch.

Reply:
The function is
```lua

-- truncate a string based on number of display columns/cells it occupies
-- so that multibyte characters are not broken up mid-character
---@param str string
---@param col_limit number
---@return string
local function truncate_by_cell(str, col_limit)
  if str and str:len() == api.nvim_strwidth(str) then return fn.strcharpart(str, 0, col_limit) end
  local short = fn.strcharpart(str, 0, col_limit)
  if api.nvim_strwidth(short) > col_limit then
    while api.nvim_strwidth(short) > col_limit do
      short = fn.strcharpart(short, 0, fn.strchars(short) - 1)
    end
  end
  return short
end
````

Reply:
Can definitely be improved haven't looked at this in while tbh but as the name suggests the idea is truncation by cell. There's also since been the addition of `vim.str_utfindex` but really not sure if that can help.